test/client.test.ts

```diff
@@ -161,6 +161,21 @@ describe('client', async () => {
       expect(() => createClient({projectId: 'abc123', logger: console})).toThrow(/logger/)
     })
 
+    test('throws on invalid perspective', () => {
+      expect(() => createClient({projectId: 'abc123', perspective: 'published'})).not.toThrow(
```

Comment:
I would just remove string from the type, and do //@ts-expect-error on these tests.